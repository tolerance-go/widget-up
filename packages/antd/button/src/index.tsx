import React, { useEffect } from "react";
import { useState } from "react";
import { Button as AntdButton } from "antd";
import { formSettings } from "widget-up-lib";

const Button = ({ settings }: { settings?: Record<string, any> }) => {
  const [text, setText] = useState("");

  useEffect(() => {
    return formSettings.on("text:changed", ({ value }) => {
      setText(value);
    });
  }, []);

  const [color, setColor] = useState("red");

  return (
    <span className="antd_v3_26_20">
      <AntdButton
        style={{
          color,
        }}
        onClick={() => {
          setColor((prev) => (prev === "red" ? "blue" : "red"));
        }}
      >
        hi button
      </AntdButton>
    </span>
  );
};

export default Button;
