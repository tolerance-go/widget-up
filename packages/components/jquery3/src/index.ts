import $ from "jquery";

const Button = () => {
  const $button = $("<button>Hi Button</button>").css({ color: "red" });

  $button.on("click", function () {
    const color = $button.css("color");
    if (color === "rgb(255, 0, 0)") {
      $button.css("color", "blue");
    } else {
      $button.css("color", "red");
    }
  });

  return $button;
};

export default Button;
export { Button as Component };
