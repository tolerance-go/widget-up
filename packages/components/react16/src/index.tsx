import React from "react";
import { useState } from "react";

const Button = () => {
  const [color, setColor] = useState("red");
  return (
    <button
      style={{
        color,
      }}
      onClick={() => {
        setColor((prev) => (prev === "red" ? "blue" : "red"));
      }}
    >
      hi button
    </button>
  );
};

export default Button;
export { Button as Component };
