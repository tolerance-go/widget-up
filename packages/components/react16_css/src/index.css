button.react16_css {
  background-color: black;
}
