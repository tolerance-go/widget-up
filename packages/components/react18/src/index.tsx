import React from "react";
import { useState } from "react";

const Button = () => {
  const [color, setColor] = useState("red");
  return (
    <button
      style={{
        color,
      }}
      onClick={() => {
        setColor((prev) => (prev === "red" ? "blue" : "red"));
      }}
    >
      hi button react18
    </button>
  );
};

export default Button;
export { Button as Component };
