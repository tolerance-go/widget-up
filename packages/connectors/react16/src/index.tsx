import { ReactElement } from "react";
import ReactDOM from "react-dom";

const render = ({
  rootElement,
  component,
}: {
  rootElement: HTMLElement;
  component: ReactElement;
}) => {
  // 确认 rootElement 存在
  if (!rootElement) {
    console.error("Root element not found.");
    return;
  }

  ReactDOM.render(component, rootElement);
};

const unmount = ({ rootElement }: { rootElement: HTMLElement }) => {
  // 尝试卸载 rootElement 上现有的 React 组件
  if (ReactDOM.unmountComponentAtNode(rootElement)) {
    console.log("Component unmounted successfully.");
  } else {
    console.log(
      "No component was mounted on rootElement, or unmount was unsuccessful."
    );
  }
};

export { render, unmount };
