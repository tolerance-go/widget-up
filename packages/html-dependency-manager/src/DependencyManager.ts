import semver from "semver";

interface DependencyDetail {
  version: string;
  subDependencies: Record<string, DependencyDetail>;
}

class DependencyManager {
  private dependencies: { [key: string]: any[] };
  private versionList: { [key: string]: string[] };

  constructor(versionList: { [key: string]: string[] }) {
    this.versionList = versionList;
    this.dependencies = {};
  }

  addDependency(
    dependency: string,
    version: string,
    subDependencies?: { [key: string]: string },
  ) {
    let resolvedVersion = this.resolveVersion(dependency, version);
    if (!resolvedVersion) return;

    // 检查是否已存在该依赖项
    if (!this.dependencies[dependency]) {
      this.dependencies[dependency] = [];
    }

    let existingDep = this.dependencies[dependency].find(
      (dep) => dep.version === resolvedVersion,
    );

    if (existingDep) {
      // 更新现有依赖的子依赖
      if (subDependencies) {
        for (let subDep in subDependencies) {
          let subVersion = this.resolveVersion(subDep, subDependencies[subDep]);
          if (subVersion) {
            existingDep.subDependencies[subDep] = { version: subVersion };
          } else {
            // 如果找不到合适的版本，可能需要移除该子依赖
            delete existingDep.subDependencies[subDep];
          }
        }
      }
    } else {
      // 添加新的依赖项
      let dependencyObject: DependencyDetail = {
        version: resolvedVersion,
        subDependencies: {},
      };

      if (subDependencies) {
        for (let subDep in subDependencies) {
          let subVersion = this.resolveVersion(subDep, subDependencies[subDep]);
          if (subVersion) {
            dependencyObject.subDependencies[subDep] = {
              version: subVersion,
              subDependencies: {},
            };
          }
        }
      }

      this.dependencies[dependency].push(dependencyObject);
    }
  }

  removeDependency(dependency: string, version: string) {
    if (!this.dependencies[dependency]) {
      return; // 如果没有找到这个依赖名称，直接返回
    }

    // 过滤掉要删除的特定版本
    this.dependencies[dependency] = this.dependencies[dependency].filter(
      (dep) => dep.version !== version,
    );

    // 如果删除后该依赖项变为空数组，则删除该属性
    if (this.dependencies[dependency].length === 0) {
      delete this.dependencies[dependency];
    }
  }

  getDependencies() {
    return this.dependencies;
  }

  private resolveVersion(
    dependency: string,
    versionRange: string,
  ): string | undefined {
    const versions = this.versionList[dependency];
    if (!versions) return undefined;

    const sortedVersions = versions.sort(semver.rcompare);
    for (let version of sortedVersions) {
      if (semver.satisfies(version, versionRange)) {
        return version;
      }
    }
    return undefined;
  }
}

export { DependencyManager };
