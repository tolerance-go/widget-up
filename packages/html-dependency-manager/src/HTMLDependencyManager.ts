import { DependencyManager, DependencyDetail } from "./DependencyManager";
import semver from "semver";

class HTMLDependencyManager {
  private dependencyManager: DependencyManager;
  private fetchVersionList: (dependencyName: string) => Promise<string[]>;
  private versionCache: { [key: string]: string[] };

  constructor(fetchVersionList: (dependencyName: string) => Promise<string[]>) {
    this.fetchVersionList = fetchVersionList;
    this.dependencyManager = new DependencyManager({});
    this.versionCache = {};
  }

  // 更新依赖版本列表，如果没有缓存
  private async updateVersionList(dependency: string) {
    if (!this.versionCache[dependency]) {
      const versions = await this.fetchVersionList(dependency);
      this.versionCache[dependency] = versions;
      this.dependencyManager.updateVersionList({ [dependency]: versions });
    }
  }

  // 收集并去重所有需要更新的依赖列表
  private async collectDependencies(
    dependency: string,
    subDependencies?: { [key: string]: string },
    collectedDependencies = new Set<string>(),
  ) {
    collectedDependencies.add(dependency);

    if (subDependencies) {
      for (const subDependency in subDependencies) {
        this.collectDependencies(
          subDependency,
          undefined,
          collectedDependencies,
        );
      }
    }

    return collectedDependencies;
  }

  // 更新所有收集到的依赖版本列表
  private async updateAllVersionLists(dependencies: Set<string>) {
    for (const dependency of dependencies) {
      await this.updateVersionList(dependency);
    }
  }

  // 递归收集所有相关依赖并更新版本列表
  async collectAndUpdateVersionLists(
    dependency: string,
    subDependencies?: { [key: string]: string },
  ) {
    const allDependencies = await this.collectDependencies(
      dependency,
      subDependencies,
    );
    await this.updateAllVersionLists(allDependencies);
  }

  async addDependency(
    dependency: string,
    versionRange: string,
    subDependencies?: { [key: string]: string },
  ): Promise<string | undefined> {
    // 确保依赖版本列表是最新的
    await this.collectAndUpdateVersionLists(dependency, subDependencies);

    // 添加主依赖项
    return this.dependencyManager.addDependency(
      dependency,
      versionRange,
      subDependencies,
    );
  }

  removeDependency(dependency: string, versionRange: string) {
    this.dependencyManager.removeDependency(dependency, versionRange);
  }

  getDependencies() {
    return this.dependencyManager.getDependencies();
  }
}

export { HTMLDependencyManager };
