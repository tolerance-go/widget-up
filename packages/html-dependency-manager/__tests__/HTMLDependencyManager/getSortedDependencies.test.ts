// HTMLDependencyManager.test.ts

import { HTMLDependencyManager } from "@/src/HTMLDependencyManager";

describe("HTMLDependencyManager getSortedDependencies", () => {
  let manager: HTMLDependencyManager;
  const mockFetchVersionList = jest.fn();

  beforeEach(async () => {
    mockFetchVersionList.mockReset();
    mockFetchVersionList.mockImplementation((dependencyName: string) => {
      const versions: Record<string, string[]> = {
        react: ["16.8.0", "16.13.1", "17.0.0"],
        "react-dom": ["16.8.0", "16.13.1", "17.0.0"],
        redux: ["4.0.4", "4.0.5", "4.1.0"],
      };
      return Promise.resolve(versions[dependencyName] || []);
    });

    manager = new HTMLDependencyManager(mockFetchVersionList);
    await manager.addDependency(
      "react",
      "^16.8.0",
      { "react-dom": "^16.8.0" },
    );
    await manager.addDependency("redux", "^4.0.5", { react: "^17.0.0" });
  });

  test("getSortedDependencies should sort dependencies correctly", () => {

    expect(manager.getDependencies()).toMatchSnapshot();

    // const sortedDependencies = manager.getSortedDependencies();

    // expect(sortedDependencies).toEqual([
    //   {
    //     name: "react-dom",
    //     version: "16.13.1",
    //     isGlobal: false,
    //     subDependencies: {},
    //   },
    //   {
    //     name: "react",
    //     version: "16.13.1",
    //     isGlobal: true,
    //     subDependencies: {
    //       "react-dom": {
    //         name: "react-dom",
    //         version: "16.13.1",
    //         isGlobal: false,
    //         subDependencies: {},
    //       },
    //     },
    //   },
    //   {
    //     name: "react",
    //     version: "17.0.0",
    //     isGlobal: false,
    //     subDependencies: {},
    //   },
    //   {
    //     name: "redux",
    //     version: "4.1.0",
    //     isGlobal: true,
    //     subDependencies: {
    //       react: {
    //         name: 'react',
    //         version: "17.0.0",
    //         isGlobal: false,
    //         subDependencies: {},
    //       },
    //     },
    //   },
    // ]);
  });
});
