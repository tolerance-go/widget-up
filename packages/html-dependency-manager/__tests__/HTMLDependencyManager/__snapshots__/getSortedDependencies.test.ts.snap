// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLDependencyManager getSortedDependencies getSortedDependencies should sort dependencies correctly 1`] = `
{
  "react": [
    {
      "isGlobal": true,
      "name": "react",
      "subDependencies": {
        "react-dom": {
          "isGlobal": false,
          "name": "react-dom",
          "subDependencies": {},
          "version": "16.13.1",
        },
      },
      "version": "16.13.1",
    },
    {
      "isGlobal": false,
      "name": "react",
      "subDependencies": {},
      "version": "17.0.0",
    },
  ],
  "react-dom": [
    {
      "isGlobal": false,
      "name": "react-dom",
      "subDependencies": {},
      "version": "16.13.1",
    },
  ],
  "redux": [
    {
      "isGlobal": true,
      "name": "redux",
      "subDependencies": {
        "react": {
          "isGlobal": false,
          "name": "react",
          "subDependencies": {},
          "version": "17.0.0",
        },
      },
      "version": "4.1.0",
    },
  ],
}
`;
